### PR TITLE
Update definition of datacenter system for power measurement

### PR DESCRIPTION
**Pull request by anirban-ghosh**
Updated networking requirements of datacenter systems to better reflect real servers.  There are 2 main changes in this PR:

- In addition to the ECC requirement already present for Datacenter systems, a networking requirement was also added.  A datacenter system must have some minimum networking and this must be described in the System JSON.  This is important when measuring system power and avoid artificially lowering system power by stripping out necessary networking equipment from the server.
- Changed the LoadGen Operation section to require all datacenter submissions to provide a high level system architecture description of their system to show that the bandwidth from the networking/IO devices is enough to sustain the data requirement for all benchmarks/scenarios.  Previously this requirement only existed for systems where the data did not originate from host CPU DRAM.  This change makes requires all submissions to prove that the path from the 'most logical source of origin' to the 'most logical DRAM' has enough bandwidth.
- For example, the above two changes combined will ensure a datacenter server submission has enough networking ingress bandwidth to the system to sustain the server scenario throughput.

### CODE REVIEW

File: inference_rules.adoc
Comment (arjunsuresh) [inline inference_rules.adoc:238]: Does populating the networking and actually using them cause a difference in power usage?

File: inference_rules.adoc
Comment (arjunsuresh) [inline inference_rules.adoc:385]: I dont think it makes sense to really enforce this requirement when loadgen is actually not using the networking interface. It would make sense if the server scenario is modified so that queries are indeed coming via the networking interface.

**Review by arjunsuresh — CHANGES_REQUESTED** [review]

File: inference_rules.adoc
Comment (arjunsuresh) [reply inference_rules.adoc:238]: Yes, we'll get the idle power usage and for network interface it may not vary significantly based on load. But it does not look good to have this requirement unless loadgen actually serves the queries over the network interface for the server scenario. Can we have this modification for loadgen server scenario and then this requirement will be trivially satisfied by all submitters?

File: inference_rules.adoc
Comment (anirban-ghosh) [reply inference_rules.adoc:385]: Right, this was a WIP and we did discuss within the Power WG that this doesn't make sense.  Instead, we decided to make some of the System JSON fields more descriptive and better suited for power.  This change will affect only power submissions, and will not affect any other submissions like perf.-only or training perf.  

I'll close this PR shortly and reopen another one once the Power WG approves the changes.  Please wait for that update and review once I create the PR.

File: inference_rules.adoc
Comment (anirban-ghosh) [reply inference_rules.adoc:238]: As I mentioned in the other comment thread - while we would like transparency into the populated components during a power submission in order to make power submissions more reviewable and transparent, we discussed in the Power WG that this was not the way to do it.  We will instead use the System JSON and file a different PR to achieve the same goal but without having to enforce bandwidth requirements or change the loadgen operation rules as this PR does.

Comment (s-idgunji) [reply inference_rules.adoc:238]: Regardless of the variation  of load or not , we submit system power measured during the performance phase, so the power from components like NICs will add to measured power.   The ask really is that you document your system components in the system Json and not about having a mandatory number of NICs .  What part of the description above does not look good. Could you copy/paste the exact part ?

Comment (arjunsuresh) [reply inference_rules.adoc:238]: Sure. The problem is with the "necessary" part. 

all the **necessary** networking required by the system architecture described 

When server scenario does not mandate network usage submitters are open to not use network at all. If **necessary** means only the components which are actually being used for the submission run (like for interacting with the power server),  the rule is fine. But if **necessary** in turn means all the networking components which are required to server the inference load to the server, that is a much bigger requirement.

Comment (s-idgunji) [reply inference_rules.adoc:238]: "A datacenter system must be equipped with all the necessary networking required by the system architecture described in the <<LoadGen Operation>> section."

@arjunsuresh  - adding the full sentence for clarity

As @anirban-ghosh  wrote, this will be discussed separately in Inference WG and through a separate PR , not tied for Power WG or power submissions.   What a data center system definition entails is a separate topic.

Comment (arjunsuresh) [reply inference_rules.adoc:238]: Thank you @s-idgunji for the clarification. I understand that what the `necessary` part is part of a separate discussion. But consider the below line: 

> All necessary networking must be populated if power is measured along with performance.

Here, the statement makes sense only if we can measure the performance without using networking. Can we avoid the confusion by renaming the section heading to **IO** and changing the description as follows? 

> A datacenter system must be equipped with all the necessary networking or IO devices as required by the system architecture described in the LoadGen Operation section. The details of the networking components must be described in the appropriate field of the System JSON (https://github.com/mlcommons/policies/blob/master/submission_rules.adoc). 

**The above change prohibits the usage of Network mounted disks**

**Comment by anirban-ghosh** [comment]
Update:

Keeping this PR open since it was discussed in the Inference WG.  Power WG is no longer involved in this PR (as per @TheKanter's comment above), so handing over to Inference WG to take this up for further discussion.

**Comment by nitinqti** [comment]
I understand the intent here and I am open to suggestions made. IMO, we need to think through before we put such definitions and constraints. 

- Primarily we need to focus on AI ML workloads, benchmarking and Power needed for these benchmarks
- We need to grow this community much larger beyond current participants (Couple accelerators and couple CPU based participants). The more we close/more constraints we pull, more difficult it becomes for new entrants. It works like barrier
- Perhaps better way to do is, keep options open for submitters. If submitter want to do ML Benchmark, allow them to do it. No need to add more peripherals that are not used by ML benchmarks. 
- If suggestions are to do real use-case based system benchmarks, introduce that as an additional category of benchmarks and keep those as optional. Those who submit, will benefit by showing prowess of their solutions. Also make a note that AI ML deployments are driven by varying use cases and TCOs and we cannot cover them all here.

To summarize my opinion: Focus on growing the submitters community, there are plenty of players in market and we need to bring them over to MLPerf. Putting constraints like this will only discourage new players to join/submit. Lets have options for submitters.

**Comment by DilipSequeira** [comment]
@nitinqti I agree entirely that real use-case benchmarks would be useful for MLPerf, but that's beyond the scope of this PR. 

Also that we need more participants, and should be careful about imposing conditions that would discourage submitters.  But MLPerf is a system benchmark, not an accelerator benchmark, and we should expect that submitted systems would be intended for production. So it's not clear why this requirement should be considered a significant discouragement. 

And requiring systems to be viable for production helps ensure that submitters are not creating unrealistic systems and that measurements are comparable.  Our goal here is to document only the basic minimum of requirements for systems to be viable, but if there easier ways to ensure that, we're happy to discuss.

**Review by arjunsuresh — COMMENTED** [review]
I'm considering Resnet50 as an example for a clarification on this BW requirement. Currently loadgen loads at least 1024 images to a buffer in memory (DDR) and queries are issued from here. So, even though images are loaded from disk (or other IO), they are stored in DDR and getting reused. Now, when we add the throughput achieved requirement of network or I/O bandwidth, should not this also add to the DDR BW requirement? i.e., should not there be sufficiently free DDR BW to support this network traffic?

**Comment by DilipSequeira** [comment]
@arjunsuresh you have to demonstrate bandwidth from the NIC to the accelerator. So if your networking goes through system DRAM in production, you would indeed have to demonstrate that you're not bottlenecked by system DRAM bandwidth. OTOH if you have a path from the NIC direct to the accelerator, you don't have this problem in production.

NMS being done on the host does not itself create additional networking bandwidth requirements. My perspective is that the PR represents a reasonable approximation of what network bandwidth is needed, but please feel free to propose a refinement if you have a concern that some submissions may be bottlenecked by host-device bandwidth in production rather than network bandwidth, and thus results may not be reflective of production environments.

**Comment by arjunsuresh** [comment]
Thank you @DilipSequeira for your clarification. As you have mentioned NMS being done on the host does not create additional networking BW requirement but do add to the DRAM BW usage. I think as of the current state we can avoid the DDR requirement and DDR and CPU requirements can be checked when eventually loadgen serves the queries though the network.  So, can we propose the networking BW requirement as a theoretical requirement and may not be practical? Say, if a PCIe 4.0 x 8 slot is free on the system, we can assume that it supports 32 GBps data transfer without worrying if there is enough CPU or DRAM capability to take care of this data transfer. Is this an acceptable approximation for simplicity?

**Comment by DilipSequeira** [comment]
For 2.0, yes - although IIRC PCIe Gen4 is 2GB/s/lane unidirectional, so you would be limited to 16GB/s each way in an 8-way slot. For 2.1 and beyond, you have to populate the slot with a sufficiently fast NIC.

If you're starting with the trace in host DRAM, then indeed the system performance will demonstrate that sufficient host-accelerator bandwidth is available. If you're using start_from_device/end_on_device, then your system architecture writeup has to demonstrate that it's available.

**Comment by rnaidu02** [comment]
Intel would like to propose the following changes to the PR for v2.1 and beyond performance only measurements:

	- Requiring a writeup (similar to v2.0 requirement) to show that there exists enough bandwidth for each NIC (and not required to have NIC present during the measurements as the presence of NIC is not going to impact any performance)

**Comment by anirban-ghosh** [comment]
> Intel would like to propose the following changes to the PR for v2.1 and beyond performance only measurements:
> 
> ```
> - Requiring a writeup (similar to v2.0 requirement) to show that there exists enough bandwidth for each NIC (and not required to have NIC present during the measurements as the presence of NIC is not going to impact any performance)
> ```

@rnaidu02 :  In the Networking section of "Minimum requirements of a datacenter system" it is specified that ` All necessary networking must be populated if power is measured along with performance.` (Link: https://github.com/mlcommons/inference_policies/blob/1f94bf36f25f1855a750b7bd90a7e7f29e55badd/inference_rules.adoc#networking).  Since it specifies that all networking must be populated if power is measured, is the above rule sufficiently to implicitly imply that when power is not measured, the components need not be populated?

**Comment by rnaidu02** [comment]
> > Intel would like to propose the following changes to the PR for v2.1 and beyond performance only measurements:
> > ```
> > - Requiring a writeup (similar to v2.0 requirement) to show that there exists enough bandwidth for each NIC (and not required to have NIC present during the measurements as the presence of NIC is not going to impact any performance)
> > ```
> 
> @rnaidu02 : In the Networking section of "Minimum requirements of a datacenter system" it is specified that ` All necessary networking must be populated if power is measured along with performance.` (Link: https://github.com/mlcommons/inference_policies/blob/1f94bf36f25f1855a750b7bd90a7e7f29e55badd/inference_rules.adoc#networking). Since it specifies that all networking must be populated if power is measured, is the above rule sufficient to implicitly imply that when power is not measured, the components need not be populated?

@anirban-ghosh If that is the case, then it is sufficient.

**Comment by rnaidu02** [comment]
Approved from Inference WG

**Review by rnaidu02 — APPROVED** [review]
Approved the changes from Inference WG

**Pull request merged**
